tests/unit_tests/client/proxy/http.cc

```diff
@@ -57,4 +57,16 @@ BOOST_AUTO_TEST_CASE(ok) {
   BOOST_CHECK(tmp.HandleData(tmpData));
 }
 
+BOOST_AUTO_TEST_CASE(jumpI2PAddressHelper) {
+  kovri::client::HTTPMessage tmp;
+  // Jump service GET parameter
+  std::string tmpData = "GET stats.i2p?i2paddresshelper=";
+  // Base64-encoded I2P address
+  tmpData+="WED~HHcaMK5GdfwrIOnDLj27Waso9~UUM4PBy~Dqn8-chmJwb1yg3l7WTsMRKkAkMs7vAk3pWOLvxZPRxGP3jP1ELNUg3mL2Dvj8euij8efF38fxzUcgNQ-DpFR40D44nn2si9COqIDjm1njixVBOy7ab4QWFSdE5DgEGOQxjkRiAU5K5vNnq6iFT~idukr1PASA3P5KWDu4s9g~S7k9FHev~HPhUArbuS6U75YlKjQXxhZhxDzdotZ~xUS2ZJcZlsH~73OO6zL3FgCs4JlDkklgbwarK57F9Auys6mRC1KkYgmavE~w6IA8kDp4IbGnhb2Rtcfb0SZoVtifVNkflvXX8bjkaNUAZtjmRKFt9I47pHSrdNfpgCV8a5kFl1QC5Li8OD3YmZg3kgX8Q0uc4IBtTzFvyT73hlIa~guoYYsstVJ58el01RPVYqJeEm1hlxloGSgm6WheFfY664IN7J92dhZvmL4DdZdIgJwXwjKM5QCcwpp2b~yYyY~XUjSCBQAEAAcAAA==";
```

Comment:
If you want a newline, you need to add a newline.

Reply:
I didn't want a newline. 

Was separating for clarity, since those three assignments to `tmpData` should all be one HTTP header line.

Reply:
On a side-note, you'll want to be mindful of websafe encoding. Example browser request:

```
GET http://kovri.i2p/?i2paddresshelper=0UVPqAA4xUSfPYPBca24h8fdokhwcJZ4zWvELv-5OsBYTHKtnLzvK7byXtXT~fOV2pExi8vrkgarGTNDfJbB2KCsdVS3V7qwtTvoCGYyklcDBlJsWMj7H763hEz5rt9SzLkcpwhO3t0Zwe6jXL1UB-QW8Kx[2/5549]
Pc6OiJ1QpnE6Bo5OUm6jPurQGXdWCAPio5Z-YnRL46n0IHWOQPYYSStJMYPlPS-S75rMIKbZbEMDraRvSzYAphUaHfvtWr2rCSPkKh3EbrOiBYiAP2oWvAQCsjouPgVF2qwQRnBbiAezHedM2gXzkgIyCV2kGOOcHhiihd~7fWwJOloH-gO78QkmCuY-3kp3633v3MBw7pmABr-XNK
WnATZOuf2syWVBZbTnOXsWf41tu6a33HOuNsMxAOUrwbu7QRmT4X8X-EITwNlKN6r1t3uoQ~yZm4RKsJUsBGfVtKl8PBMak3flQAg95oV0OBDGuizIQ9vREOWvPGlQCAXZzEg~cUNbfBQAEAAcAAA%3d%3d HTTP/1.1
Host: kovri.i2p
User-Agent: Mozilla/5.0 (Windows NT 6.1; rv:52.0) Gecko/20100101 Firefox/52.0
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate
DNT: 1
Connection: keep-alive
Upgrade-Insecure-Requests: 1
```

Notice the encoded `==` sent from the browser. Though this doesn't necessarily apply to the test-case at the moment, just an FYI.

Reply:
Great catch, I wasn't even thinking about URL encoding.

It doesn't look like the HTTP proxy is doing any kind of encoding/decoding at the moment.

Should I try to implement URI processing for the HTTP proxy in another PR, using Boost.Beast or cpp-netlib? If so, I can add it to #812 for tracking.